src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImportAliasCommand.cs

```diff
@@ -290,103 +378,118 @@ private Collection<AliasInfo> GetAliasesFromFile(bool isLiteralPath)
         {
             Collection<AliasInfo> result = new Collection<AliasInfo>();
 
+            // it seems like filePath is always null and never defined, so we can remove it as far as I could see
+            // bug keeping it for now because tests are failing
             string filePath = null;
             using (StreamReader reader = OpenFile(out filePath, isLiteralPath))
             {
-                CSVHelper csvHelper = new CSVHelper(',');
-
-                Int64 lineNumber = 0;
+                long lineNumber = 0;
                 string line = null;
                 while ((line = reader.ReadLine()) != null)
                 {
                     ++lineNumber;
 
-                    // Ignore blank lines
-                    if (line.Length == 0)
+                    if (LineShouldBeSkipped(line)) 
                     {
                         continue;
                     }
 
-                    // Ignore lines that only contain whitespace
-                    if (OnlyContainsWhitespace(line))
-                    {
-                        continue;
-                    }
+                    Collection<string> parsedLine = ParseCsvLine(line);
 
-                    // Ignore comment lines
-                    if (line[0] == '#')
+                    // create options
+                    KeyValuePair<ScopedItemOptions, bool> optionPair = CreateItemOptions(parsedLine, filePath, lineNumber);
+                    ScopedItemOptions options = optionPair.Key;
+                    if (IsValidParsedLine(parsedLine, optionPair.Value, lineNumber, filePath)) 
                     {
-                        continue;
+                        result.Add(ConstructAlias(parsedLine, options));
                     }
+                }
 
-                    Collection<string> values = csvHelper.ParseCsv(line);
+                reader.Dispose();
+            }
+            
+            return result;
+        }
 
-                    if (values.Count != 4)
-                    {
-                        string message = StringUtil.Format(AliasCommandStrings.ImportAliasFileInvalidFormat, filePath, lineNumber);
+        private KeyValuePair<ScopedItemOptions, bool> CreateItemOptions(Collection<string> parsedLine, string filePath, long lineNumber) 
+        {
+            ScopedItemOptions options = ScopedItemOptions.None;
+            bool succesfullParse = false;
+            try 
+            {
+                succesfullParse = Enum.TryParse<ScopedItemOptions>(parsedLine[3], out options);
```

Comment:
If we use TryParse we need to remove `try-catch` and use `if` statement.

Reply:
`TryParse<TEnum>(String, TEnum) is identical to the Parse(Type, String) method, except that instead of throwing an exception, it returns false if the conversion fails. It eliminates the need for exception handling when parsing the string representation of an enumeration value.`

I guess that means I'll have to remove the `WriteError` part since that takes an exception and these are not thrown anymore. That will change functionality a bit, is that a problem? 
Is it fine to just skip (so let it return false) instead of writing an error?

Reply:
If I remove the argException it becomes impossible to pass the test regarding `Should throw an error when reading more than four values`. 

As far as I can see: using `if` and skipping `try/catch` is not possible without failing this test since it expects an exception.

Is it okay to skip the `tryParse` and switch it to a normal `Parse`, or do you have a different suggestion?

```
Should throw an error when reading more than four values 41ms
Expected an exception, with FullyQualifiedErrorId 'ImportAliasFileFormatError,Microsoft.PowerShell.Commands.ImportAliasCommand' to be thrown, but no exception was thrown.
```

Reply:
Our preference is to avoid exceptions if possible. If an API allows this we should use it. Changing error reporting is not a breaking change. I think the test will remain unchanged (using -Error Stop will cause the exception)